File: src/main/java/nuclearcoder/discordbot/database/SqlUsers.java
package nuclearcoder.discordbot.database;

import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class SqlUsers {

    public static final void createUser(String userID) throws SQLException
    {
        PreparedStatement statement = Database.conn
                .prepareStatement("INSERT INTO `users` (`uid`) VALUES (?)");

        statement.setString(1, userID);

        statement.executeUpdate();

        statement.close();
    }

    public static final boolean isUserPresent(String userID) throws SQLException
    {
        PreparedStatement statement = Database.conn
                .prepareStatement("SELECT NULL FROM `users` WHERE `uid` = ?");

        statement.setString(1, userID);

        ResultSet rs = statement.executeQuery();
        boolean isPresent = rs.first();

        rs.close();
        statement.close();

        return isPresent;
    }

    public static final void setOperator(String userID, int value) throws SQLException
    {
        PreparedStatement statement = Database.conn
                .prepareStatement("UPDATE `users` SET `operator` = ? WHERE `uid` = ?");

        statement.setInt(1, value);
        statement.setString(2, userID);

        statement.executeUpdate();

        statement.close();
    }

    public static final boolean isOperator(String userID) throws SQLException
    {
        PreparedStatement statement = Database.conn
                .prepareStatement("SELECT `operator` FROM `users` WHERE `uid` = ?");

        statement.setString(1, userID);

        ResultSet rs = statement.executeQuery();
        boolean isOperator = rs.first() && (rs.getInt(1) != 0);

        rs.close();
        statement.close();

        return isOperator;
    }

}

File: src/main/java/nukebot/database/SqlUsers.kt
package nukebot.database

import java.sql.SQLException

object SqlUsers {

    @Throws(SQLException::class)
    fun createUser(userID: String) {
        val statement = Database.conn
                ?.prepareStatement("INSERT INTO `users` (`uid`) VALUES (?)") ?: return

        statement.setString(1, userID)

        statement.executeUpdate()

        statement.close()
    }

    @Throws(SQLException::class)
    fun isUserPresent(userID: String): Boolean {
        val statement = Database.conn
                ?.prepareStatement("SELECT NULL FROM `users` WHERE `uid` = ?") ?: return false

        statement.setString(1, userID)

        val rs = statement.executeQuery()
        val isPresent = rs.first()

        rs.close()
        statement.close()

        return isPresent
    }

    @Throws(SQLException::class)
    fun setOperator(userID: String, value: Int) {
        val statement = Database.conn?.prepareStatement("UPDATE `users` SET `operator` = ? WHERE `uid` = ?") ?: return

        statement.setInt(1, value)
        statement.setString(2, userID)

        statement.executeUpdate()

        statement.close()
    }

    @Throws(SQLException::class)
    fun isOperator(userID: String): Boolean {
        val statement = Database.conn
                ?.prepareStatement("SELECT `operator` FROM `users` WHERE `uid` = ?") ?: return false

        statement.setString(1, userID)

        val rs = statement.executeQuery()
        val isOperator = rs.first() && rs.getInt(1) != 0

        rs.close()
        statement.close()

        return isOperator
    }

}
